src/content/development/contribution-guide/_index.en.md

```diff
@@ -10,23 +10,30 @@ in order for their contributions to the project to be addressed in a timely mann
 
 ## Project Resources
 
-To be expanded
+Submariner uses [GitHub Projects](https://docs.github.com/en/issues/organizing-your-work-with-project-boards) to manage releases. Read
+“[Tracking progress on your project board](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards)"
+to learn more on how to work with projects.
 
 ### Backlog Board
 
-To be expanded
+[The backlog board](https://github.com/orgs/submariner-io/projects/15) hosts issues or features that are not scheduled
+for the release currently being developed.
 
 ### Current Release Board
 
-To be expanded
+Current release work is tracked in [the 0.13 board](https://github.com/orgs/submariner-io/projects/20).
 
 ### Enhancements Repository
 
-To be expanded
+All enhancement proposals need to be submitted to the [enhancements repository](https://github.com/submariner-io/enhancements). To submit a
+new enhancement proposal, raise an 
+[Enhancement Request issue](https://github.com/submariner-io/enhancements/issues/new?assignees=&labels=enhancement&template=enhancement.md)
+on the repository.
 
 ### Releases Repository
 
-To be expanded
+Submariner's release is automated to a great extent. [The release process documentation](https://submariner.io/development/release-process/)
+explains the details of a release. The release automation is maintained in the [releases repository](https://github.com/submariner-io/releases).
```

Comment:
You should add details that all submariner releases (containing the subctl binaries) can be found here.